### ISSUE TITLE
Improve interface for Download::Negotiate to allow alternatives to Prefer::SortVersions

### ISSUE THREAD

**Issue by plicease** [Perl5-Alien/Alien-Build#43]
`Prefer::SortVersions` is only IMO only sort of okay, but it probably covers 95% of all cases.  That said it would be nice for `Download::Negotiate` to make it easier to customize the sort for that 5% of cases.  Right now you can do this like thus:

```perl
use alienfile;

probe sub { 'share' };

share {

  start_url 'https://github.com/premake/premake-core/releases';
  plugin 'Download';
  plugin 'Decode::HTML';

  meta->register_hook(prefer => sub {
    my($build, $res) = @_;
    ...
  });

};
```

I would like to improve this by adding a `prefer` property that can be one of:

 - the value 0 which indicates not to add a prefer hook or plugin.  It is the responsibility of the `alienfile` to add either a hook or plugin.
 - a code reference which will be used as the prefer hook.
 - <strike>a plugin name of the form `Prefer::OtherPreferPlugin`.</strike>

Like the `version` property the `prefer` property should imply the use of an appropriate decode plugin.

Examples:

```perl
share {
  start_url '...';
  plugin Download => (
    prefer => 0,
  );
  plugin 'Prefer::OtherPreferPlugn';
};
```
```perl
share {
  start_url '...';
  plugin Download => (
    prefer => 0,
  );
  meta->register_hook(
    prefer => sub {
      my($build, $res) = @_;
      ...
    },
  );
};
```
```perl
# xxx do not implment
share {
  start_url '...';
  plugin Download => (
    prefer => 'Prefer::OtherPreferPlugn',
  );
};
```
```perl
share {
  start_url '...';
  plugin Download => (
    prefer => sub {
      my($build, $res) = @_;
      ...
    },
  );
};
```

appropriate tests should be added to `t/alien_build_plugin_download_negotiate.t`.

My thinking about this came up in the discussion on #42.

**Comment by plicease** [comment]
Never mind on setting prefer to a plugin.  That interface doesn't really make sense.

**Comment by plicease** [comment]
This is implemented in 1.30
